src/runtime/tuple.h

```diff
@@ -93,11 +100,38 @@ static inline tuple get_tuple(value e, symbol a)
     return (v && is_tuple(v)) ? v : 0;
 }
 
+/* get and validate that result is a vector type */
+static inline vector get_vector(value e, symbol a)
+{
+    value v = get(e, a);
+    return (v && is_vector(v)) ? v : 0;
+}
+
+/* get and return if vector, else convert from tuple */
+static inline vector get_vector_or_convert_tuple(value e, symbol a)
+{
+    value v = get(e, a);
+    if (v) {
+        if (is_vector(v))
+            return v;
```

Comment:
That makes sense, and I went about changing this function to create a new vector before copying the content of the old one over. However, I noticed that there is a flaw in the tuple conversion whereby the copying would stop if the key number space had a hole in it (was sparse). But, in attempting to solve that, I couldn't think of a way to avoid bad input (say, some huge index attribute) from resulting in a huge allocate while at the same time supporting a tuple with sparse entries.

Given that this function is used in only two places, I decided to remove it entirely and replace the existing calls with iterates. Since it's possible to iterate over a tuple and vector using the same code, and also because we will eventually deprecate the use of tuples-as-vectors, I see no reason to try and keep this tuple-to-vector conversion any longer.